src/System.Management.Automation/engine/regex.cs

```diff
@@ -55,6 +55,12 @@ public sealed class WildcardPattern
         //
         private const char escapeChar = '`';
 
+        //
+        // Threshold for stack allocation.
+        // The size is less than MaxShortPath = 260.
+        //
+        private const int StackAllocThreshold = 256;
```

Comment:
Why using `MaxShortPath` as the bar? We are using different thresholds at different places for `stackalloc` operations.

From SMA:
https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/Utils.cs#L208
https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/FormatAndOutput/common/BaseOutputtingCommand.cs#L255

Maybe there are more in other dlls.

Reply:
It depends on a context. The code can be involved for completions - it is very short strings, and for path globbing - common case is paths with lengths < MaxShortPath.
(We need a telemetry if we want see more real data.)

Reply:
corefx code doesn't have a single threshold either, so it should be fine to use different threshold in our code too, I guess.